### PR TITLE
Improve Login Flow

### LINKED ISSUE #304: Fix login screen button
**When you click the login button on the login screen with lab dropdown menu, it takes a bit to open the website to enter your credentials. In this time you can press the button a few more times causing confusing behaviour.**

Maybe deactivate button behaviour after one press (activate it again when it is necessary)
An easy workaround would be a short delay i guess

### PR DESCRIPTION
**Pull request by Benjamin-Frost**
Closes #304

- [x] I have read the [code-style guidelines](https://github.com/hpi-dhc/PharMe/blob/main/docs/FLUTTER_STYLE.md) and confirmed that this PR upholds said guidelines

### CODE REVIEW

File: app/lib/common/constants.dart [hunk 1]
```diff
@@ -7,7 +7,10 @@ final annotationServerIp = Uri.https(
 );
 final annotationServerUrl = annotationServerIp.replace(path: 'api/v1');
 final labServerUrl = labServerIp.replace(path: 'api/v1');
-final keycloakUrl = labServerIp;
+final keycloakUrl = Uri.https(
+  'keycloak.k8s.kunst.me',
+  '',
+);
 final cpicMaxCacheTime = Duration(days: 90);
```
Comment (SebastianWagner2) [inline app/lib/common/constants.dart:14]: Not sure we should put this in the repo but now it already is i guess

Did you ask Samuel about this?

File: app/lib/login/pages/cubit.dart [hunk 2]
```diff
@@ -16,10 +18,25 @@ class LoginPageCubit extends Cubit<LoginPageState> {
...
+          authUrl: lab.authUrl,
+          tokenUrl: lab.tokenUrl,
+        );
+      } on PlatformException catch (e) {
+        if (e.code == 'CANCELED') {
+          revertToInitialState();
+        } else {
+          emit(LoginPageState.error(context.l10n.err_generic));
+        }
+        return;
```
Comment (SebastianWagner2) [inline app/lib/login/pages/cubit.dart:38]: You could show the specific error from below here or just don't

**Review by SebastianWagner2 — CHANGES_REQUESTED** [review]
Just more specific error message

**Comment by Benjamin-Frost** [comment]
Also, rename `endpoint` to something more specific and make it an `Uri` instead of a `String`

File: app/lib/common/constants.dart (hunk 1 above)
Comment (Benjamin-Frost) [reply app/lib/common/constants.dart:14]: I guess it should be fine o.0.

**Review by SebastianWagner2 — APPROVED** [review]

**Pull request merged**